tap_google_search_console/client.py

```diff
@@ -1,312 +1,107 @@
 import json
 from datetime import datetime, timedelta
+from typing import Any
 from urllib.parse import quote
 import backoff
 import requests
 from requests.exceptions import Timeout, ConnectionError
-
+from .exceptions import (
+    raise_for_error,
+    Server5xxError,
+    GoogleQuotaExceededError,
+    GoogleRateLimitExceeded,
+)
 import singer
+from datetime import timezone
 from singer import metrics
 from singer import utils
 
-BASE_URL = 'https://www.googleapis.com/webmasters/v3'
-GOOGLE_TOKEN_URI = 'https://oauth2.googleapis.com/token'
+BASE_URL = "https://www.googleapis.com/webmasters/v3"
+GOOGLE_TOKEN_URI = "https://oauth2.googleapis.com/token"
 LOGGER = singer.get_logger()
 
 # set default timeout of 300 seconds
 REQUEST_TIMEOUT = 300
 
-class GoogleError(Exception):
-    pass
-
-class Server5xxError(GoogleError):
-    pass
-
-class GoogleBadRequestError(GoogleError):
-    pass
-
-class GoogleUnauthorizedError(GoogleError):
-    pass
-
-class GooglePaymentRequiredError(GoogleError):
-    pass
-
-class GoogleNotFoundError(GoogleError):
-    pass
-
-class GoogleMethodNotAllowedError(GoogleError):
-    pass
-
-class GoogleConflictError(GoogleError):
-    pass
-
-class GoogleGoneError(GoogleError):
-    pass
-
-class GooglePreconditionFailedError(GoogleError):
-    pass
-
-class GoogleRequestEntityTooLargeError(GoogleError):
-    pass
-
-class GoogleRequestedRangeNotSatisfiableError(GoogleError):
-    pass
-
-class GoogleExpectationFailedError(GoogleError):
-    pass
-
-class GoogleForbiddenError(GoogleError):
-    pass
-
-class GoogleUnprocessableEntityError(GoogleError):
-    pass
-
-class GooglePreconditionRequiredError(GoogleError):
-    pass
-
-class GoogleRateLimitExceeded(GoogleError):
-    pass
-
-class GoogleInternalServiceError(Server5xxError):
-    pass
-
-class GoogleNotImplementedError(Server5xxError):
-    pass
 
-class GoogleServiceUnavailable(Server5xxError):
-    pass
+class GoogleClient:  # pylint: disable=too-many-instance-attributes
+    def __init__(self, client_id: str, client_secret: str,
+                 refresh_token: str, site_urls: str, user_agent=None,
+                 timeout=REQUEST_TIMEOUT):
```

Comment:
we should pass the config directly to client

Reply:
What's the harm in doing like this? 
I may have to skip doing this change because it requires lot more changes in unit test cases